app/controllers/api/v1/chats_controller.rb

```
@@ -0,0 +1,17 @@
+# frozen_string_literal: true
+
+class Api::V1::ChatsController < ApplicationController
+  before_action :authenticate_user!, only: [:finish]
+
+  def finish
+    chat = Chat.find(params[:chat_id])
+    chat.state = 2 # finished
+    if chat.save
+      render json: chat, status: :ok
+    else
+      render json: chat.errors.messages, status: :not_modified
+    end
+  rescue ActiveRecord::RecordNotFound
```

This could be verified in the code why let it throw an exception for something easily verifiable?

---

the exception is correctly catched with rescue and returns an appropiate message. Petter told me it was a good practice to do it in that way. Do you think adding an 'if chat.nil?' statement is cleaner?